yt_dlp/extractor/porn91.py

```diff
@@ -32,29 +57,27 @@ def _real_extract(self, url):
         if '作为游客，你每天只可观看10个视频' in webpage:
             raise ExtractorError('91 Porn says: Daily limit 10 videos exceeded', expected=True)
 
-        title = self._search_regex(
-            r'<div id="viewvideo-title">([^<]+)</div>', webpage, 'title')
-        title = title.replace('\n', '')
+        title = self._html_extract_title(webpage)
+        title = remove_end(title.replace('\n', ''), 'Chinese homemade video').strip()
 
         video_link_url = self._search_regex(
-            r'<textarea[^>]+id=["\']fm-video_link[^>]+>([^<]+)</textarea>',
+            r'document\.write\(\s*strencode2\s*\(\s*((?:"[^"]+\")|(?:\'[^\']+\'))\s*\)\s*\)',
             webpage, 'video link')
-        videopage = self._download_webpage(video_link_url, video_id)
-
-        info_dict = self._parse_html5_media_entries(url, videopage, video_id)[0]
+        video_link_url = self._search_regex(
+            r"src=\'([^\']+)\'", urllib.parse.unquote(video_link_url), 'unquoted video link')
 
         duration = parse_duration(self._search_regex(
-            r'时长:\s*</span>\s*(\d+:\d+)', webpage, 'duration', fatal=False))
+            r'时长:\s*<span[^>]*>\s*(\d+:\d+)\s*</span>', webpage, 'duration', fatal=False))
 
         comment_count = int_or_none(self._search_regex(
-            r'留言:\s*</span>\s*(\d+)', webpage, 'comment count', fatal=False))
+            r'留言:\s*<span[^>]*>\s*(\d+)\s*</span>', webpage, 'comment count', fatal=False))
 
-        info_dict.update({
+        return {
             'id': video_id,
+            'url': video_link_url,
+            'ext': 'mp4',
```

Comment:
This is incorrect as well. There are some examples of `video_link_url` with `m3u8` extension. I'm looking at `_parse_html5_media_entries` to fill in the value unless there is something easier.

Reply:
https://github.com/yt-dlp/yt-dlp/blob/990dd7b00fc5f7c22ff339d0ca5b5b4f21923dca/yt_dlp/extractor/airtv.py#L63-L69 may be helpful

Reply:
Perfect, thanks!